src/hotspot/cpu/riscv/templateTable_riscv_64.cpp

```diff
@@ -2555,155 +2547,147 @@ void TemplateTable::getfield_or_static(int byte_no, bool is_static, RewriteContr
 
   __ align(32, 28, 28); // Align load.
   // __ bind(Ldtos);
-  __ Assembler::fence(Assembler::RW_OP, Assembler::RW_OP); // Volatile entry point (one instruction before non-volatile_entry point).
+  //__ Assembler::fence(Assembler::RW_OP, Assembler::RW_OP); // Volatile entry point (one instruction before non-volatile_entry point).
+  __ nop();
   assert(branch_table[dtos] == 0, "can't compute twice");
   branch_table[dtos] = __ pc(); // non-volatile_entry point
   __ add(R30_TMP5, Rclass_or_obj, Roffset);
   __ fld(F23_ftos, R30_TMP5, 0);
   __ push(dtos);
   if (!is_static && rc == may_rewrite) {
-    patch_bytecode(Bytecodes::_fast_dgetfield, Rbc, Rscratch);
+    //patch_bytecode(Bytecodes::_fast_dgetfield, Rbc, Rscratch); FIXME_RISCV
   }
   {
-    Label acquire_double;
     __ bnez(Rscratch, Lisync); // Volatile?
     __ dispatch_epilog(vtos, Bytecodes::length_for(bytecode()));
-
-    __ bind(acquire_double);
-    __ fcmpu_PPC(CCR0, F23_ftos, F23_ftos); // Acquire by cmp-br-isync.
-    __ beq_predict_taken_PPC(CCR0, Lisync);
-    __ b_PPC(Lisync); // In case of NAN.
   }
 
   __ align(32, 28, 28); // Align load.
   // __ bind(Lftos);
-  __ Assembler::fence(Assembler::RW_OP, Assembler::RW_OP); // Volatile entry point (one instruction before non-volatile_entry point).
+  //__ Assembler::fence(Assembler::RW_OP, Assembler::RW_OP); // Volatile entry point (one instruction before non-volatile_entry point).
+  __ nop();
```

Comment:
Could you describe, why is this correct?

Reply:
There is no barrier here in arm. I have no idea why. And I have no idea why it's needed in ppc.